Formula/p/poetry.rb

```diff
@@ -200,6 +195,9 @@ class Poetry < Formula
   end
 
   def install
+    # poetry > filelock > hatchling, fix to `ZIP does not support timestamps before 1980` error
+    ENV["SOURCE_DATE_EPOCH"] = Time.now.to_i.to_s
```

Comment:
looks like upstream issue? 

another interesting thing to note is poetry 1.8.5 does not need it

Reply:
So brew sets `SOURCE_DATE_EPOCH` based on the [max mtime](https://github.com/Homebrew/brew/blob/b6fafba4864d65acabf966415e14b2dd86d81e1a/Library/Homebrew/build.rb#L146) of all the formula's files. With this poetry release, the mtimes are now all `0`, i.e. 1970

This bug was already reported to hatch at https://github.com/pypa/hatch/issues/1605 but closed as won't fix. Maybe they would be willing to reconsider

Reply:
> With this poetry release, the mtimes are now all `0`, i.e. 1970

any reason why this happened?

Reply:
For reproducible builds, relevant change is https://github.com/python-poetry/poetry-core/pull/766